cclient/http/socket.c

```diff
@@ -5,168 +5,156 @@
  * Refer to the LICENSE file for licensing information
  */
 
-#include <stdbool.h>
-#include <stdio.h>
-#include <stdlib.h>
 #include <string.h>
 
 #include "socket.h"
 
-int socket_connect(char const *const hostname, const size_t port) {
-  struct addrinfo hints, *serverinfo, *info;
-  char port_string[6] = {};
-  int sockfd = -1;
-
-  memset(&hints, 0, sizeof hints);
-  hints.ai_family = AF_UNSPEC;  // use AF_INET6 to force IPv6
-  hints.ai_socktype = SOCK_STREAM;
-  sprintf(port_string, "%zu", port);
-  if (getaddrinfo(hostname, port_string, &hints, &serverinfo) != 0) {
-    return -1;
+// socket receiving timeout
+#define MBEDTLS_SOCKET_TIMEOUT 30000
+
+static void mbedtls_conf_init(mbedtls_ctx_t *tls_ctx, bool is_https) {
```

Comment:
I feel like functions prefix with `mbedtls_` make me difficult to distinguish actual mbedtls functions and our wrappers. An abstraction should abbreviate to what it does imho. What do you think?

Reply:
sorry about that, done.